### ISSUE TITLE
CodeQuality: Doxygen regex doesn't work for a warning that's on two lines

### ISSUE THREAD

**Issue by lit-af** [melexis/warnings-plugin#135]
Hi,

I found a small bug with the Doxygen regex, the code-quality report generated description field is incomplete when there a warning message that's on two lines.

DoxygenWarningsLog.txt
```text
C:/DEV/kog-ipc-common/src/kog_ipc.c:485: warning: The following parameter of Ipc_handleReceivedMessage(KogIpc_t *pIpc, IpcMessageEnvelope_t *pMessage, bool *pDisposeEnvelope) is not documented:
  parameter 'pDisposeEnvelope'
```

Doxygen_quality.json
```json
    {
        "severity": "major",
        "location": {
            "path": "kog-ipc-common\\src\\kog_ipc.c",
            "lines": {
                "begin": 485
            }
        },
        "description": "The following parameter of Ipc_handleReceivedMessage(KogIpc_t *pIpc, IpcMessageEnvelope_t *pMessage, bool *pDisposeEnvelope) is not documented:",
        "fingerprint": "504e0ad956f6bf387431e0737ce0708a"
    },
```

I'm wondering if the issue comes from my doxygen configuration or from the regex.

**Comment by lit-af** [comment]
Ok, I figured out what the regex should be!

```python
r"(?:(?P<path1>(?:[/.]|[A-Za-z]).+?):(?P<line1>-?\d+):\s*(?P<severity1>[Ww]arning|[Ee]rror)|<.+>:(?P<line2>-?\d+)(?::\s*(?P<severity2>[Ww]arning|[Ee]rror))?): (?P<description1>.+(?:(?!\s*([Nn]otice|[Ww]arning|[Ee]rror): )[^/<\n][^:\n][^/\n].+)*[\s\n]*[^/<\n][^:\n][^/\n].+)|\s*\b(?P<severity3>[Nn]otice|[Ww]arning|[Ee]rror): (?!notes)(?P<description2>.+)\n?"
```

**Comment by Letme** [comment]
Great  - can you open a Pull Request? You have a test case there so it should be quite straight forward and we welcome new contributions.

**Comment by lit-af** [comment]
I'd be willing to, but there's very little documentation on how to contribute and I'm mainly a C/C++ dev.

Is there any chance you could add instructions on how to add/run tests and how to run a development version of the package?

**Comment by lit-af** [comment]
I haven't been able to test this locally as I'm getting an error when when trying to install locally. I tested my [regex online](https://regex101.com/r/xuJA2C/) using the `doxygen_warnings.txt` log found in this repo.

**Comment by Letme** [comment]
That's great feedback - I will look into it along with the fact that we might be missing this installation/test guidelines in `CONTRIBUTING.md` file (you can help to improve as well).

Repo has `tox` for setting up your environment and installing stuff (so you simply run `tox` after you installed it), but it seems that is is not working at them moment since we relied on Continious Integration for too long (and also that is not running from your Pull Request). I will look into that as well in next few days.

**Comment by lit-af** [comment]
Thanks for actively responding to issues raised here. It's quite refreshing to see an OSS repo with active support for once.

I was able to install the package in a docker container running ubuntu 22.04. I figured out that `git` is a dependency, but it's not mentionned in the install instructions.

**Comment by Letme** [comment]
Yes, probably because if you manage to clone the repo, you should have git installed. Most of the time you should be using virtual environments (`python -m venv venv` and then `source venv/bin/activate`) for Python development, but I also use docker for keeping the C development.

We are glad to help. Since we use all our tools daily, then most of the time it is really cool to see other people using them. Now I just need to find some time :sweat_smile:

**Comment by JasperCraeghs** [comment]
Your changes in PR #136 probably require changes to the reference/input files of the unit tests.

**Comment by lit-af** [comment]
I actually cloned your repo's master branch to make sure that I was able to run the tests first.
